lib/Request.js

```diff
@@ -179,22 +179,31 @@ class Request extends CallableInstance {
 
   async _setup (opts = {}) {
     try {
-      const connection = await this._remit._connection
-      this._channel = await connection.createChannel()
+      if (!this._remit._publishChannels[opts.event]) {
+        const publishChannelP = this._remit._publishChannels[opts.event] = (async () => {
+          const connection = await this._remit._connection
+          return connection.createChannel()
+        })()
+
+        const publishChannel = await publishChannelP
+        
+        await publishChannel.consume(
+          'amq.rabbitmq.reply-to',
+          bubble.bind(this._incoming.bind(this)),
+          {
+            noAck: true,
+            exclusive: true
+          }
+        )
+      }
+  
+      this._channel = await this._remit._publishChannels[opts.event]
+
       this._channel.on('error', console.error)
```

Comment:
I think this even and the `close` event below should be moved in to the creation too.

Reply:
yeah that's probably a leak otherwise!